skyportal/handlers/api/source.py

```diff
@@ -288,3 +293,101 @@ def get(self, source_id):
         if not set(source.groups).intersection(set(self.current_user.groups)):
             return self.error('Inadequate permissions.')
         return self.success(data={'photometry': source.photometry})
+
+
+class SourceOffsetsHandler(BaseHandler):
+    @auth_or_token
+    def get(self, source_id):
+        """
+        ---
+        description: Retrieve offset stars to aid in spectroscopy
+        parameters:
+        - in: path
+          name: source_id
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: facility
+          nullable: true
+          schema:
+            type: string
+            enum: [null, Keck, Shane, P200]
+          description: Which facility to generate the starlist for
+        - in: query
+          name: how_many
+          nullable: true
+          schema:
+            type: integer
+            minimum: 0
+            maximum: 10
+          description: |
+            Requested number of offset stars (set to zero to get starlist
+            of just the source itself)
+        - in: query
+          name: obstime_isoformat
+          nullable: True
+          schema:
+            type: string
+          description: |
+            datetime of observation in isoformat (e.g. 2020-12-30T12:34:10)
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        source = Source.query.get(source_id)
```

Comment:
If we should be checking user permissions here, this can become 
`source = Source.get_if_owned_by(source_id, self.current_user)`

Reply:
Fixed, although note that I used the query pattern in other source-related APIs. Might need to change some other parts of the apis to conform to this.